pkg/kv/kvserver/rangefeed: BenchmarkRangefeedBudget failed

pkg/kv/kvserver/rangefeed.BenchmarkRangefeedBudget [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachprodMicrobenchmarkWeeklyWip/20685401?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachprodMicrobenchmarkWeeklyWip/20685401?buildTab=artifacts#./artifacts/microbench/experiment) on master @ [cc61b3c16062dfeef5ee82e6a80d868d499e047e](https://github.com/cockroachdb/cockroach/commits/cc61b3c16062dfeef5ee82e6a80d868d499e047e):

Fatal error:

```
panic: unknown rangefeed test type

```
Stack: 

```
goroutine 55 gp=0xc000602f00 m=3 mp=0xc0000d3008 [running]:
panic({0x22fc640?, 0x2d31af0?})
	GOROOT/src/runtime/panic.go:802 +0x168 fp=0xc0005e7880 sp=0xc0005e77d0 pc=0x49f2a8
github.com/cockroachdb/cockroach/pkg/kv/kvserver/rangefeed.newTestProcessor({0x2d8ce60, 0xc000710588}, {0xc0005e7d80, 0x5, 0x0?})
	pkg/kv/kvserver/rangefeed/processor_helpers_test.go:457 +0x870 fp=0xc0005e7ae8 sp=0xc0005e7880 pc=0x1f54b50
github.com/cockroachdb/cockroach/pkg/kv/kvserver/rangefeed.runBenchmarkRangefeed(0xc000710588, {{0x0, 0x0}, {0x27e7421, 0x5}, 0x1, 0x0})
	pkg/kv/kvserver/rangefeed/bench_test.go:92 +0x2c5 fp=0xc0005e7e98 sp=0xc0005e7ae8 pc=0x1f3fc05
github.com/cockroachdb/cockroach/pkg/kv/kvserver/rangefeed.BenchmarkRangefeedBudget.func1(0xc000710588?)
	pkg/kv/kvserver/rangefeed/bench_test.go:71 +0x75 fp=0xc0005e7f10 sp=0xc0005e7e98 pc=0x1f3f915
testing.(*B).runN(0xc000710588, 0x1)
	GOROOT/src/testing/benchmark.go:219 +0x190 fp=0xc0005e7fa0 sp=0xc0005e7f10 pc=0x59ab10
testing.(*B).run1.func1()
	GOROOT/src/testing/benchmark.go:245 +0x48 fp=0xc0005e7fe0 sp=0xc0005e7fa0 pc=0x59b188
runtime.goexit({})
	src/runtime/asm_amd64.s:1693 +0x1 fp=0xc0005e7fe8 sp=0xc0005e7fe0 pc=0x4a7d81
created by testing.(*B).run1 in goroutine 29
	GOROOT/src/testing/benchmark.go:238 +0x90
```
<details><summary>Log preceding fatal error</summary>
<p>

```
goos: linux
goarch: amd64
cpu: Intel(R) Xeon(R) CPU @ 2.60GHz
BenchmarkRangefeedBudget
BenchmarkRangefeedBudget/budget=false

```
</p>
</details>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/kv-triage
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*BenchmarkRangefeedBudget.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-56167

--- a/pkg/kv/kvserver/rangefeed/bench_test.go
+++ b/pkg/kv/kvserver/rangefeed/bench_test.go
@@ -63,17 +63,20 @@ func BenchmarkRangefeed(b *testing.B) {
 // processes a set of events.
 func BenchmarkRangefeedBudget(b *testing.B) {
 	for _, budget := range []bool{false, true} {
-		b.Run(fmt.Sprintf("budget=%t", budget), func(b *testing.B) {
-			var budgetSize int64
-			if budget {
-				budgetSize = math.MaxInt64
-			}
-			runBenchmarkRangefeed(b, benchmarkRangefeedOpts{
-				opType:           writeOpType,
-				numRegistrations: 1,
-				budget:           budgetSize,
+		for _, procType := range testTypes {
+			b.Run(fmt.Sprintf("budget=%t/proc=%s", budget, procType), func(b *testing.B) {
+				var budgetSize int64
+				if budget {
+					budgetSize = math.MaxInt64
+				}
+				runBenchmarkRangefeed(b, benchmarkRangefeedOpts{
+					rangefeedTestType: procType,
+					opType:            writeOpType,
+					numRegistrations:  1,
+					budget:            budgetSize,
+				})
 			})
-		})
+		}
 	}
 }
 

